fw/http_limits.c

```diff
@@ -627,6 +627,40 @@ frang_assert_host_header(const TfwStr *l, const TfwStr *r)
 	return tfw_strcmp(l, r);
 }
 
+static int
+frang_http_domain_fronting_check(const TfwHttpReq *req, FrangAcc *ra)
+{
+	TlsCtx *tctx;
+
+	if ((TFW_CONN_TYPE(req->conn) & TFW_FSM_HTTPS) == 0)
+		return TFW_PASS;
+
+	tctx = tfw_tls_context(req->conn);
+
+	if (tctx->vhost != req->vhost) {
+		TfwVhost *tls_vhost = tctx->vhost;
+		BasicStr tls_name, req_name;
+		static BasicStr null_name = {"NULL", 4};
+
+		/* An exotic case where TLS connection hasn't assigned
+		 * any vhost to the TlsCtx */
+		if (unlikely(tls_vhost == NULL))
+			return TFW_PASS;
```

Comment:
`tfw_tls_sni` has this lines:
```
/* No SNI => no vhost for later frang checks */
ctx->vhost = NULL;
```
but in this place we return `TFW_PASS` in case of `tctx->vhost` is `NULL`. This implies anyone can not send ServerName extension or send empty SNI and avoid any checks. @RomanBelozerov Do we have tests for such cases?

Reply:
>This implies anyone can not send ServerName extension or send empty SNI and avoid any checks.
I'd like to emphasize that SNI is not a part of server-side protection mechanism. It is required  in order to sent client certificate for SSL-enabled vhosts that share same IP.
For me such behavior is a perfect example of an idiom "all elements of an empty set are green because it has no elements". Missing SNI is always valid as long as client is ready to accept certificate from default vhost.